### DISCUSSION TITLE
consult-line on certain file yields error

### DISCUSSION THREAD

**Discussion by drcxd** [minad/consult#1114] [Q&A] [answered]
I am not sure if this is consult's problem, can you help me check? 

Part of the stack trace:

```
Debugger entered--Lisp error: (error "Error in syntax_table logic for intervals <-")
  font-lock-fontify-keywords-region(1507 177296 nil)
  font-lock-default-fontify-region(1507 177296 nil)
  font-lock-fontify-region(1507 177296)
  #f(compiled-function (fun) #<bytecode -0x155074056fe5fe07>)(font-lock-fontify-region)
  jit-lock--run-functions(1507 177296)
  jit-lock-fontify-now()
  consult--fontify-all()
  consult--line-candidates(nil 3)
  #f(compiled-function () #<bytecode -0x10796e074ed8a12c>)()
  consult-line(nil nil)
```

I can only reproduce this on a certain file, which has around 4000 lines and I can not locate which part triggers this error. Unfortunately I can not share the file directly since it contains propretary content. 

My consult package is on 07ea542. 

My Emacs version is GNU Emacs 29.4 (build 2, x86_64-w64-mingw32) of 2024-07-05.

**Comment by minad** [comment, marked as answer]
See https://github.com/minad/consult?tab=readme-ov-file#bug-reports for some guidance. The bug most lies most likely in the fontification of the major mode you are using.
